### PR TITLE
Parquet output, issue #75

### PR DESCRIPTION
**Pull request by sgomezg**
#### Description
Output for parquet files. It closes issue #75. We need an extra parameter ```path``` and optional one ```datePattern``` if you want path formatting. I changed joda-time by NScala-time (Scala wrapper). Also I changed a tailrec function to improve readability. Added methods to DateOperations to calculate paths with patterns and granularity.

#### Reviewer
@arinconstrio 

#### Test
Added 7 test. All passed

#### Scalastyle
All test passed.

### CODE REVIEW

**Comment by arinconstrio** [comment]
I think the example policy json file  is missed

File: plugins/output-parquet/src/main/scala/com/stratio/sparkta/plugin/output/parquet/ParquetOutput.scala [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  override val multiplexer = Try(properties.getString("multiplexer").toBoolean).getOrElse(false)
+
+  override val isAutoCalculateId = Try(properties.getString("isAutoCalculateId").toBoolean).getOrElse(false)
+
+  final val Slash: String = "/"
+
+  override def upsert(dataFrame: DataFrame, tableName: String): Unit = {
+    val path = properties.getString("path", None)
+    require(path.isDefined)
+    dataFrame.save(s"${path.get}$Slash$tableName$subPath", "parquet", SaveMode.Append)
```
Comment (gserranojc) [inline plugins/output-parquet/src/main/scala/com/stratio/sparkta/plugin/output/parquet/ParquetOutput.scala]: The output mode must be Overwrite.

**Comment by sgomezg** [comment]
Changes done. Ping @arinconstrio @gserranojc

**Comment by arinconstrio** [comment]
Looks good for me, @gserranojc ?

File: plugins/output-parquet/src/main/scala/com/stratio/sparkta/plugin/output/parquet/ParquetOutput.scala [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  override val supportedWriteOps = Seq(WriteOp.Inc, WriteOp.IncBig, WriteOp.Set, WriteOp.Max, WriteOp.Min,
+    WriteOp.AccAvg, WriteOp.AccMedian, WriteOp.AccVariance, WriteOp.AccStddev, WriteOp.FullText, WriteOp.AccSet)
+
+  override val multiplexer = Try(properties.getString("multiplexer").toBoolean).getOrElse(false)
+
+  override val isAutoCalculateId = Try(properties.getString("isAutoCalculateId").toBoolean).getOrElse(false)
+
+  override def upsert(dataFrame: DataFrame, tableName: String): Unit = {
+    val path = properties.getString("path", None)
+    require(path.isDefined)
```
Comment (gserranojc) [inline plugins/output-parquet/src/main/scala/com/stratio/sparkta/plugin/output/parquet/ParquetOutput.scala]: Could you add a logging message?

File: plugins/output-parquet/src/main/scala/com/stratio/sparkta/plugin/output/parquet/ParquetOutput.scala [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    val subPath = DateOperations.subPath(timeName, properties.getString("datePattern", None))
+    dataFrame.save(s"${path.get}/$tableName$subPath", "parquet", SaveMode.Overwrite)
+  }
+}
```
Comment (gserranojc) [inline plugins/output-parquet/src/main/scala/com/stratio/sparkta/plugin/output/parquet/ParquetOutput.scala]: Could you add the object with the function to insert configuration in sparkContext. This parameter is important:
("spark.sql.parquet.binaryAsString", "true")

Equal than Cassandra output.

Comment (gserranojc) [inline plugins/output-parquet/src/main/scala/com/stratio/sparkta/plugin/output/parquet/ParquetOutput.scala]: Please add the information in outputs documentation.

**Comment by sgomezg** [comment]
Changes done. Ping @gserranojc

**Pull request merged**